lib/urlwatch/cli.py

```diff
@@ -98,13 +98,12 @@ def main():
                                    config_file, urls_file, hooks_file, cache_file, False)
     setup_logger(command_config.verbose)
 
-    # setup storage API
-    config_storage = YamlConfigStorage(command_config.config)
-    cache_storage = CacheMiniDBStorage(command_config.cache)
-    urls_storage = UrlsYaml(command_config.urls)
+    # migration
+    migrate_urls(command_config)
+    migrate_cache(command_config)
```

Comment:
Previously before the migration, `check_directories()` was called, which created directories on demand -- this is now done AFTER migration, isn't that a problem?

Reply:
Looking at the logic of these two migration functions, I don't think this would be a problem. Migration happens only if `urls_txt`/`cache_dir` exist, but they are both under `urlwatch_dir` by definition.

In any case, I've added a commit that creates parent directory of `urls` and `cache` before the migration code creates these files.